SPA_howto.md

```diff
@@ -0,0 +1,206 @@
+
+#### Add a single record manually
+
+The "+" icon enables the user to **Add a single new data record**, to the current data table. Upon clicking it, the user is presented with a form where the values for each field of the new record can be entered. 
+
+![SPA_addsingle.png](figures/SPA_addsingle.png)
+
+If invalid data has been entered and the user attempts to save that invalid data, validation error messages are shown as in the example above when we edited and existing record. 
+
+Once you finish typing the data click on the "save" icon. You don't need to add data to all fields, but you will be asked if you are sure you want to leave some fields blank. Click "yes" to proceed:
+
+![SPA_addblankok.png](figures/SPA_addblankok.png)
+
+Your new record will be saved. You can click on the "table" icon on the top to see the country table again:
+
+![SPA_addsucess.png](figures/SPA_addsucess.png)
+
+
+
+#### Add several records from a file
+
+Adding single records one by one is useful sometimes, but many users want to add data in bulk. Users often have data in tables that were created in MS Excel, recorded with a digital device or by any other mean. You can import this data into Zendro from an **Excel file** (.xlsx), a **comma separated value file** (.csv) or a **json** file. Here we will cover the first two.
+
+The .xlsx or .csv data file is expected to follow these requirements:
+
+1. Column names in the first row must correspond to model attributes (i.e. column names in the Zendro main table for that model)
+2. Empty values should be represented as `NULL`.
+3. Date and time formats must follow the [RFC 3339](https://tools.ietf.org/html/rfc3339) standard.
+
+Additionally, if you are uploading your data from a csv file:
+
+1. String (text) records should be quoted by `"`. For example `"Ingredient A, Ingredient B"` instead of `Ingredient A, Ingredient B`. However if there are no commas (`,`) within any single record, then the quotes are not necessary. 
+2. `"NULL"` should be quoted.
+
+In order to get the field names right and check what type each one is (e.g. integer, character, etc), you can **Download the model template** by clicking in the "light down arrow" at the top right panel of the main data model table. You will be prompted to download a csv file named after the table you are downloading, for example "river". You can open this file in your favourite spreadsheet processor (e.g. Excel).
+
+It will have the the column names you need in the first row, and the data type in the second. Notice that if your data model has associations, the ids ("key") associating records will be shown as columns too. For example because the model "river" is associated with one or more "countries" you have to provide the `country_ids` in the last column. If we fill this example with a list of Mexican rivers, this means that the `country_ids` field should include Mexico's country id, which is "MX" according to the "country" data model table.
```

Comment:
@wunderbarr  is currently working on changing that, so we should make sure to also update this part of the documentation

Reply:
Yes. Marked.